src/cordova/platform/addHelper.js

```diff
@@ -292,7 +292,9 @@ function getVersionFromConfigFile (platform, cfg) {
 function downloadPlatform (projectRoot, platform, version, opts) {
     var target = version ? (platform + '@' + version) : platform;
     return Q().then(function () {
-        if (opts.fetch) {
+        var platformPath = path.join(projectRoot, 'platforms', platform);
+        var platformAlreadyAdded = fs.existsSync(platformPath);
+        if ((opts.fetch) && (!platformAlreadyAdded || platformAlreadyAdded === undefined || platformAlreadyAdded === null)) {
```

Comment:
If platformAlreadyAdded is undefined or null the first condition will be truthy so there is no point in adding the additional tests.

Reply:
Okay. Thank you! I will remove it.